benchmarks/misc/tests/assorted/misc-apply-array.js

```diff
@@ -0,0 +1,45 @@
+// Since bug 1108290 the JIT optimizes fun.apply(..., array) by taking
+// a fast inline path when the array is without holes and its length
+// equals its initialized length.
+//
+// This program applies arrays whose length equals their initialized
+// length, thus triggering the optimization.
+//
+// Contrast the running time of this program with the running time of
+// misc-apply-array-headroom.js, which applies arrays whose length
+// exceeds their initialized length.  That program should run in 10x
+// to 40x the time of this one.
+
+function assertEq(result, expected) {
+  if (result !== expected)
+    throw "Assertion: Expected " + expected + ", got " + result;
+}
+
+var iter = 10000;
```

Comment:
That's probably OK.  I set it high to reduce noise in comparison with the other one, but I understand the constraint.